slsim/lens.py

```diff
@@ -430,23 +412,27 @@ def deflector_magnitude(self, band):
     def point_source_arrival_times(self):
         """Arrival time of images relative to a straight line without lensing. Negative
         values correspond to images arriving earlier, and positive signs correspond to
-        images arriving later.
-
-        :return: arrival times for each image [days]
-        :rtype: numpy array
-        """
-        lens_model_list, kwargs_lens = self.deflector_mass_model_lenstronomy()
-        lens_model = LensModel(
-            lens_model_list=lens_model_list,
-            cosmo=self.cosmo,
-            z_lens=self.deflector_redshift,
-            z_source=self.source_redshift,
-        )
-        x_image, y_image = self.point_source_image_positions()
-        arrival_times = lens_model.arrival_time(
-            x_image, y_image, kwargs_lens=kwargs_lens
-        )
-        return arrival_times
+        images arriving later. This is for single source.
+
+        :return: list of arrival times for each image [days] for each source.
+        :rtype: list of numpy array
+        """
+        lens_model_list, kwargs_lens =  self.deflector_mass_model_lenstronomy()
+        image_pos = self.point_source_image_positions()
+        arrival_times_list = []
+        for index, source in enumerate(self.source):
+            lens_model = LensModel(
+                lens_model_list=lens_model_list,
+                cosmo=self.cosmo,
+                z_lens=self.deflector_redshift,
+                z_source=source.redshift,
```

Comment:
we have to feed it z_source_convention otherwise this does not work and the lens model parameters are interpreted to the specific source redshift (which it is not, in general)